[BUG] Exception is thrown when finding `productRef` instead of `fileRef`

**Describe the bug**

Exception is thrown when finding `productRef` instead of `fileRef`.


```
BAF0D4CB2AD6AE14007AC334 /* FacebookLogin in Frameworks */ = {isa = PBXBuildFile; productRef = BAF0D4CA2AD6AE14007AC334 /* FacebookLogin */; };
```


Exception is thrown when trying to remove unrelated files:

```
Error: An unexpected error occurred in clean_translation_files: 'PBXBuildFile' object has no attribute 'fileRef'
Traceback (most recent call last):
  File "/home/omar/work/openedx/openedx-app-ios/i18n_scripts/translation.py", line 394, in <module>
    main()
  File "/home/omar/work/openedx/openedx-app-ios/i18n_scripts/translation.py", line 390, in main
    clean_translation_files()
  File "/home/omar/work/openedx/openedx-app-ios/i18n_scripts/translation.py", line 335, in clean_translation_files
    xcode_project.remove_files_by_path(file_ref.path, tree=LOCALIZABLE_FILES_TREE)
  File "/home/omar/.virtualenvs/openedx-app-ios/lib/python3.10/site-packages/pbxproj/pbxextensions/ProjectFiles.py", line 366, in remove_files_by_path
    if self.remove_file_by_id(file_ref.get_id(), target_name=target_name):
  File "/home/omar/.virtualenvs/openedx-app-ios/lib/python3.10/site-packages/pbxproj/pbxextensions/ProjectFiles.py", line 332, in remove_file_by_id
    if build_file.fileRef == file_ref.get_id():
AttributeError: 'PBXBuildFile' object has no attribute 'fileRef'
make: *** [Makefile:3: clean_translations] Error 1
```


**System information** 
1. pbxproj version used: 4.1.0
2. python version used: 3.10.12
3. Xcode version used: I don't know, I use the version on https://github.com/openedx/openedx-app-ios/blob/b2539a6f0ee00a06879f3820d438430a4346363e/Core/Core.xcodeproj/project.pbxproj#L1-L8 and I have no XCode on my Linux machine

**To Reproduce**
Steps to reproduce the behavior:

TBD: Add more details here

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
The file is added despite `productRef` exists somewhere else.

**Screenshots**
None.

**Additional context**
We're trying to add generated files, and XCode doesn't cooperate without `mod-pbxproj` help.

**Related issues** 

 - https://github.com/kronenthaler/mod-pbxproj/issues/179
 - https://github.com/openedx/openedx-app-ios/pull/465

--- a/pbxproj/pbxextensions/ProjectFiles.py
+++ b/pbxproj/pbxextensions/ProjectFiles.py
@@ -329,7 +329,7 @@ class ProjectFiles:
             for build_file_id in filter(lambda x: x in self.objects, build_phase.files):
                 build_file = self.objects[build_file_id]
 
-                if build_file.fileRef == file_ref.get_id():
+                if hasattr(build_file, 'fileRef') and build_file.fileRef == file_ref.get_id():
                     # remove the build file from the phase
                     build_phase.remove_build_file(build_file)
 
@@ -339,7 +339,11 @@ class ProjectFiles:
                 target.remove_build_phase(build_phase)
 
         # remove it iff it's removed from all targets or no build file reference it
-        if len([1 for x in self.objects.get_objects_in_section('PBXBuildFile') if x.fileRef == file_ref.get_id()]) != 0:
+        if len([
+            True
+            for x in self.objects.get_objects_in_section('PBXBuildFile')
+            if hasattr(x, 'fileRef') == file_ref.get_id()
+        ]) != 0:
             return True
 
         # remove the file from any groups if there is no reference from any target

